android/src/main/kotlin/es/antonborri/home_widget/HomeWidgetPlugin.kt

```diff
@@ -73,7 +80,12 @@ class HomeWidgetPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
                     val value = prefs.all[id] ?: defaultValue
 
                     if(value is Long) {
-                        result.success(java.lang.Double.longBitsToDouble(value))
+                        // The reason for falling back to Double is that it was supported before Long.
+                        when (prefs.getString(ACTUAL_WIDGET_DATA_TYPE, "Double")) {
```

Comment:
This Key was added because there is no way to distinguish between Double and Long.

Reply:
I believe the old logic should have already covered the logic. Looking at the code I also think that there is no benefit in this extra logic. 
home_widget already handled this behavior by storing handling doubles always as long. so in that way we trade 2 bytes per number for the behavior.

So can you please revert your changes to the initial fix (adding `is Long` to the setWidgetData call)

Reply:
Why do you use longBitsToDouble to convert a Long to a Double? Also, if I want a Long (int in Dart) to be returned, is that ok?